app/src/main/java/com/example/sharingang/NewItemFragment.kt

```diff
@@ -23,12 +24,27 @@ class NewItemFragment : Fragment() {
 
         val binding: FragmentNewItemBinding =
             DataBindingUtil.inflate(inflater, R.layout.fragment_new_item, container, false)
+        var category : Int = 0
+        binding.categorySpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
+            override fun onItemSelected(
+                adapterView: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long
+            ) {
+                if(adapterView != null){category = position}
+            }
 
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
+            }
+        }

```

Comment:
I think it was better how you did in EditItemFragment. Is it possible to do the same here? (remove this and use `binding.categorySpinner.selectedItemPosition`)

Reply:
and yup I did that